src/controllers/auth.controller.js

```diff
@@ -1,11 +1,131 @@
-import {generateJWT} from '../auth/jwt';
-import {StatusCodes, ReasonPhrases} from 'http-status-codes';
+import { generateJWT, generateJwtExpiryDate } from '../auth/jwt';
+import UnauthorizedError from '../errors/UnauthorizedError'
 
-export default function googleAuth(req, res){
+export const authController = ({ config, jwt, logger, userService }) => {
+    const googleAuth = async (req, res, next) => {
+        const DEFAULT_ACCESS_TOKEN_TTL = config.get('security:jwt:access_token_ttl')
+        const DEFAULT_ACCESS_TOKEN_TTL_IN_SEC = config.get('security:jwt:access_token_ttl_in_sec')
+        const DEFAULT_REFRESH_TOKEN_TTL = config.get('security:jwt:refresh_token_ttl')
+        const DEFAULT_REFRESH_TOKEN_TTL_IN_SEC = config.get('security:jwt:refresh_token_ttl_in_sec')
 
-    //Sets jwt from authenticated google user
-    res.cookie('jwt', generateJWT(req.user))
+        if(!req || (req && !req.user)) {
+            logger.log(logger.LOGLEVEL.ERROR, { filename: __filename, logMessage: 'User not found in request' })
+            next(new UnauthorizedError('Login error'))
+            return
+        }
+    
+        // generate login hash from username
+        const user = req.user
+     
+        // Sets access and refresh token cookies and loginHash cookie from authenticated google user
+        const accessTokenCookieOptions = {
+            secure: false,
+            httpOnly: true,
+        }
+        const refreshTokenCookieOptions = {
+            secure: false,
+            httpOnly: true,
+        }
+        if(config.get('app:node_env')=== 'prod') {
+            accessTokenCookieOptions.secure = true
+            refreshTokenCookieOptions.secure = true
+        }
+        const accessToken = await generateJWT(user, DEFAULT_ACCESS_TOKEN_TTL)
+        res.cookie('accessToken', accessToken, accessTokenCookieOptions)
+    
+        const refreshToken = await generateJWT(user, DEFAULT_REFRESH_TOKEN_TTL)
```

Comment:
In theory the refresh token does not need to be a JWT. Given it just needs to be a unique random string of chars that can be exchanged for an access token if the JWT is invalid. Can chat more about this if it would help

Reply:
@ryanolee But is there any advantage of generating the refresh token just as a random string instead of jwt? If not as jwt, we still need someway to make sure that the token has not been tampered with from the client, so some kind of hashing / key signature will have to be used, which means it won’t really speed up things even without jwt.  However, with refresh token generated as jwt, we can easily set a separate expiry for refresh token that’s different from the access token expiry time? Or are there any advantage of generating refresh token using other methods?

Reply:
@ryanolee Change implemented in new push as requested.